File: app/src/main/java/org/tvheadend/tvhclient/util/billing/BillingUpdatesListener.java
package org.tvheadend.tvhclient.util.billing;

import com.android.billingclient.api.BillingClient;
import com.android.billingclient.api.Purchase;

import java.util.List;

public interface BillingUpdatesListener {

    void onBillingClientSetupFinished();

    void onConsumeFinished(String token, @BillingClient.BillingResponse int result);

    void onPurchaseSuccessful(List<Purchase> purchases);

    void onPurchaseCancelled();

    void onPurchaseError(int errorCode);
}

File: app/src/main/java/org/tvheadend/tvhclient/util/billing/BillingUpdatesListener.kt
package org.tvheadend.tvhclient.util.billing

import com.android.billingclient.api.BillingClient
import com.android.billingclient.api.Purchase

interface BillingUpdatesListener {

    fun onBillingClientSetupFinished()

    fun onConsumeFinished(token: String, @BillingClient.BillingResponseCode result: Int)

    fun onPurchaseSuccessful(purchases: List<Purchase>)

    fun onPurchaseCancelled()

    fun onPurchaseError(errorCode: Int)
}
